### ISSUE TITLE
identity macOS 13 Catalyst

### ISSUE THREAD

**Issue by polymerchm** [devicekit/DeviceKit#214]
Self-explanatory

**Comment by Zandor300** [comment]
You can do that in the following way:

```swift
#if targetEnvironment(macCatalyst)
    let catalyst = true
#else
    let catalyst = false
#endif
```

Or do you rather want a function like so?

```swift
Device.current.isMacCatalyst
```

**Comment by gabors** [comment]
Hi @Zandor300 Thanks for this awesome project!

As per above

`Device.current.description` returns `Simulator (iOS)` when running in a Mac Catalyst app.

**Comment by phimage** [comment]
`identifier` return "x86_64" when using catalyst app

so 
```

simulator(mapToDevice(identifier: ProcessInfo().environment["SIMULATOR_MODEL_IDENTIFIER"] ?? "iOS"))
```
 return `Simulator (iOS)`

---

`case mac` must be added?
The difficulty is to respond correctly to some device attribute that changed
maybe a 
`case macCatalyst(mac)` with mac an enum for all mac devices or a struct with mac definition like screen size etc...

**Comment by gabors** [comment]
Identification of Mac models would be ideal but I’m not sure what’s the apple API for that. 
For starters the description should probably fall back to a string like Mac if running under catalyst.

**Comment by phimage** [comment]
I make a PR for mac runtime and models https://github.com/devicekit/DeviceKit/pull/217

With some change I test it with mac catalyst inside an iOS app
My method to get device model identifier work on catalyst (ie. getting model identifier using `Darwin.sysct` )

So 
S1 / we could add all  mac models into enum devices 
S2 / or add a `macCatalyst(MacModel)`
with MacModel a new enum available only with iOS runtimes, which contains the list of mac Models

**Comment by Zandor300** [comment]
@phimage I think we don't need a `macCatalyst(macModel)` enum since the code that references the Mac models, in this case, will only be ran under an iOS/Catalyst target.

In case the code is in a framework that supports both iOS/Catalyst and native macOS, that can be checked using the following code I've posted here before:

```swift
#if targetEnvironment(macCatalyst)
    let catalyst = true
#else
    let catalyst = false
#endif
```

**Comment by phimage** [comment]
Yes but we could also identify a device without using `DeviceKit`

First I thinks it is crappy to return `Simulator (iOS)` on mac Catalyst when using `Device.current`, just `macCatalyst` at least

If we want to display the device name or we want to have a generic `switch case` code that rely on `DeviceKit`, making a special case each time using the `#if targetEnvironment` is annoying
